woocommerce/woocommerce-ios
# Blaze: Fix incorrect site URL sent to dashboard section

Closes: #11187 


## Description

Currently, we are using the default value `ServiceLocator.stores.sessionManager.defaultSite` for the `siteURL` property in both Blaze section on the dashboard screen and the campaign list view. 

When switching sites, `defaultSiteID` is updated before `defaultSite`, and the dashboard screen is updated before the updated value for `defaultSite` is set, causing the Blaze section to keep the outdated URL of the selected site.

This PR fixes this issue by removing the default value for the site URL, and instead injecting the site URL into the dashboard screen, which will then be used to inject into the Blaze section. This makes sure that the section always has the most up-to-date value for the selected site URL.

## Testing instructions

- Turn off the feature flag `blazei3NativeCampaignCreation`.
- Log in to a store eligible for Blaze. 
- Select the Promote button on the Blaze section in the My Store screen. Confirm that the correct product list is displayed for the selected store.
- Switch to a different store that's also eligible for Blaze.
- Select the Promote button again. Confirm that the correct product list is displayed for the selected store.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/8637727f-b27a-4a34-bc45-6b23f6b0b3b2



---
